src/utils/FormInputHandling.ts

```diff
@@ -2,14 +2,36 @@ import { instance } from "./RootClient";
 import { DbTypes } from "@/types/ResponseTypes";
 import { formatDateForPicker } from "./TimeFormatter";
 import { formatTimeForPicker } from "./TimeFormatter";
+import { off } from "process";
 
 // get data from the db when session id is generated
-async function getData(currentPage: number, limit: number) {
+async function getData(
+  currentPage: number,
+  limit: number,
+  string_query: string | undefined
+) {
   const offset = currentPage * limit;
-  const { data } = await instance.get<DbTypes[]>(`api`)
-  const hasMore = data.length > limit;
-  const items = hasMore ? data.slice(0, -1) : data;
+  const { data } = await instance.get<DbTypes[]>(`api`);
 
+  //  search in the db for the specific keyword
+  let query = string_query == undefined ? "" : string_query.toLowerCase();
+
+  const filteredData = data.filter((item) => {
+
+    // Accessing nested properties of DbTypes
+    let batch_name = item.meta_data?.batch ? item.meta_data?.batch : "";
```

Comment:
interaction with backend can be better. we have a `db-service` that allows for better filtering using batch IDs. maybe iterating over all sessions is not a good idea.

https://github.com/avantifellows/db-service/blob/main/docs/INSTALLATION.md

for now dont worry about it